### PR TITLE
Fix resolv.conf reset on machine restart

### PR DESCRIPTION
**Pull request by alexandre-slp**
After Ubuntu version 18, [Netplan](https://netplan.io/) started being responsible for managing the network configurations. The `/etc/resolv.conf` is overwritten every time the machine is restarted resulting in missing the docker-dns container ip from the nameservers list.

To solve this problem the package `resolvconf` must be installed (new dependency) and the docker-dns container ip and option must be saved in the correct files: `/etc/resolvconf/resolv.conf.d/head` and `/etc/resolvconf/resolv.conf.d/tail`

### CODE REVIEW

File: src/install.py [hunk 1]
```diff
@@ -30,56 +33,68 @@ def update_cache():
...
-            RESOLVCONF_DATA = f"options timeout:1 #@docker-dns\nnameserver {dns} #@docker-dns\n{RESOLVCONF_DATA}"
-            open(RESOLVCONF, 'w').write(RESOLVCONF_DATA)
+        name_servers = f'nameserver {dns} #@docker-dns\n' \
+                       f'nameserver 1.1.1.1 #cloudflare\n' \
+                       f'nameserver 8.8.8.8 #google\n'
+        options = 'options timeout:1 #@docker-dns\n'
+        if OS.FLAVOR == 'ubuntu' and config.OS_VERSION >= 18 * 1000:
+            open(RESOLVCONF_HEAD, 'a').write(name_servers)
+            open(RESOLVCONF_TAIL, 'a').write(options)
+            subprocess.run(['sudo', 'resolvconf', '-u'])
```
Comment (zanaca) [inline src/install.py]: Please do not use any shell process

**Review by zanaca — CHANGES_REQUESTED** [review]

File: src/install.py (hunk 1 above)
Comment (alexandre-slp) [reply src/install.py]: I am afraid there is no other way =/
This is the best alternative I have found.

File: src/install.py [hunk 2]
```diff
@@ -30,29 +33,43 @@ def update_cache():
...
-            RESOLVCONF_DATA = f"options timeout:1 #@docker-dns\nnameserver {dns} #@docker-dns\n{RESOLVCONF_DATA}"
-            open(RESOLVCONF, 'w').write(RESOLVCONF_DATA)
+        name_servers = f'nameserver {dns} #@docker-dns\n' \
+                       f'nameserver 1.1.1.1 #cloudflare\n' \
+                       f'nameserver 8.8.8.8 #google\n'
+        options = 'options timeout:1 #@docker-dns\n'
+        if OS.FLAVOR == 'ubuntu' and config.OS_VERSION >= 18 * 1000:
+            open(RESOLVCONF_HEAD, 'a').write(name_servers)
+            open(RESOLVCONF_TAIL, 'a').write(options)
+            subprocess.run(['resolvconf', '-u'])
```
Comment (alexandre-slp) [inline src/install.py:56]: Removed `sudo`.
(still working)

File: src/uninstall.py [hunk 3]
```diff
@@ -30,13 +29,20 @@ def main(name=config.DOCKER_CONTAINER_NAME, tag=config.DOCKER_CONTAINER_TAG, tld
...
+        os.remove(docker_daemon_file)
+
+    resolvconf_head = '/etc/resolvconf/resolv.conf.d/head'
+    resolvconf_tail = '/etc/resolvconf/resolv.conf.d/tail'
+    try:
+        lines = open(resolvconf_head).readlines()
+        comments = [line for line in lines if line.startswith('#')]
+        open(resolvconf_head, 'w').writelines(comments)
+        open(resolvconf_tail, 'w').write('')
+        subprocess.run(['resolvconf', '-u'])
```
Comment (alexandre-slp) [inline src/uninstall.py:43]: removed `sudo` from here too.

**Review by alexandre-slp — COMMENTED** [review]
Is it ok now?

File: src/install.py (hunk 2 above)
Comment (alexandre-slp) [reply src/install.py:56]: The best approach I have found is using `resolvconf`... Couldn't find anything better and scalable for the end user.

**Pull request merged**